src/features/common.rs

```diff
@@ -67,4 +53,159 @@ pub struct Options {
+}
+
+#[cfg(feature = "common")]
+thread_local! {
+    pub(in crate::features) static RUNTIME: runtime::Runtime<'static,'static> = runtime::Runtime::new();
+}
+
+#[cfg(feature = "common")]
+pub(in crate::features) mod runtime {
+    use std::sync::Once;
+
+    // Based on https://github.com/Valerioageno/ssr-rs/blob/main/src/ssr.rs
+    pub(in crate::features) struct Runtime<'s,'i> {
+        isolate: *mut v8::OwnedIsolate,
+        handle: *mut v8::HandleScope<'s,()>,
+        context: *mut v8::Local<'s, v8::Context>,
+        scope: *mut v8::ContextScope<'i, v8::HandleScope<'s,v8::Context>>,
+    }
+
+    impl Drop for Runtime<'_, '_> {
+        fn drop(&mut self) {
+            use std::mem::drop;
+            unsafe {
+                drop(Box::from_raw(self.scope));
+                drop(Box::from_raw(self.context));
+                drop(Box::from_raw(self.handle));
+                drop(Box::from_raw(self.isolate));
+            };
+        }
+    }
+
+    #[allow(unused)]
+    impl<'s, 'i> Runtime<'s, 'i> where 's: 'i, {
+        pub(in crate::features) fn new() -> Self {
+            #[cfg(not(feature = "external-v8"))]
+            static V8_INIT: Once = Once::new();
+
+            #[cfg(not(feature = "external-v8"))]
+            V8_INIT.call_once(|| {
+                let platform = v8::new_default_platform(0, false).make_shared();
+                v8::V8::initialize_platform(platform);
+                v8::V8::initialize();
+            });
+
+            let isolate = Box::into_raw(Box::new(v8::Isolate::new(v8::CreateParams::default())));
+            let handle = unsafe { Box::into_raw(Box::new(v8::HandleScope::new(&mut *isolate))) };
+            let context = unsafe { Box::into_raw(Box::new(v8::Context::new(&mut *handle))) };
+            let scope = unsafe { Box::into_raw(Box::new(v8::ContextScope::new(&mut *handle, *context))) };
+
+            return Runtime {isolate, handle, context, scope};
+        }
+
+        pub(in crate::features) fn get_isolate(&self) -> &mut v8::OwnedIsolate {
+            return unsafe {&mut *self.isolate};
+        }
+
+        pub(in crate::features) fn get_handle(&self) -> &mut v8::HandleScope<'s,()> {
+            return unsafe {&mut *self.handle};
+        }
+
+        pub(in crate::features) fn get_context(&self) -> &mut v8::Local<'s, v8::Context> {
+            return unsafe {&mut *self.context};
+        }
+
+        pub(in crate::features) fn get_scope(&self) -> &mut v8::ContextScope<'i, v8::HandleScope<'s,v8::Context>> {
+            return unsafe {&mut *self.scope};
+        }
+
+        pub(in crate::features) fn run<S: AsRef<str>>(&self, s: S) -> Option<v8::Local<'s, v8::Value>> {
+            let scope = self.get_scope();
+            let script = v8::String::new(scope, s.as_ref())?.into();
+            let script = v8::Script::compile(scope, script, None)?;
+            return script.run(scope);
+        }
+    }
+
+    pub(in crate::features) trait UsingRuntime<'x,'s,'i> {
```

Comment:
Trait used by `with_v8!`. This allows using a plain `Runtime` or `LocalKey<Runtime>` without an explicit `.with(|runtime| {...})`. Technically, there is no need for a trait; anything with a matching method signature will work, but this offers better semantics and prevents having to use `LocalKey`'s `with` as the name.